make/autoconf/jdk-options.m4

```diff
@@ -478,15 +478,22 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_STATIC_BUILD],
 #
 AC_DEFUN_ONCE([JDKOPT_SETUP_JMOD_OPTIONS],
 [
-  AC_ARG_WITH(jmod-compress, [AS_HELP_STRING([--with-jmod-compress],
-    [specify alternative JMOD compression])])
-  AC_MSG_CHECKING([for JMOD compression])
-  if test "x$with_jmod_compress" != "x"; then
-    JMOD_COMPRESS="$with_jmod_compress"
-    AC_MSG_RESULT([$JMOD_COMPRESS])
+  # Final JMODs are recompiled often during development, and java.base JMOD
+  # includes the JVM libraries. In release mode, prefer to compress JMODs fully.
+  # In debug mode, pay with a little extra space, but win a lot of CPU time back
+  # with the lightest (but still some) compression.
+  if test "x$DEBUG_LEVEL" = xrelease; then
+    DEFAULT_JMOD_COMPRESS="zip-6"
   else
-    AC_MSG_RESULT([default])
+    DEFAULT_JMOD_COMPRESS="zip-1"
   fi
+
+  UTIL_ARG_WITH(NAME: jmod-compress, TYPE: string,
```

Comment:
The `string` type does not check values. I am not at a computer right now so I can check, but I think you need the type called `literal`. Try greping for VALID_VALUES to see other usages.

Reply:
Indeed, it does not. Changed to `literal`.